### ISSUE TITLE
Managing multiple configuration files

### ISSUE THREAD

**Issue by vardaofthevalier** [gc3-uzh-ch/elasticluster#223]
According to the documentation on Read the Docs:

> In case the directory ~/.elasticluster/config.d exists (or, if you run with -c <PATH>, the directory <PATH>.d), all files named *.conf contained in that directory are read and parsed. In this way, you can handle multiple clusters easily by distributing the configuration over multiple files, and disable only some of them by renaming the files.

However, this doesn't appear to be working (unless I've misunderstood something).  I currently have my .elasticluster directory set up like this:

```
~/.elasticluster
    config.d/
        cluster1.conf
        cluster2.conf
```

But when I run "elasticluster list", a template configuration is written to ~/.elasticluster/config, and the files in the config.d directory are completely ignored.  I'm wondering what it is that I'm doing wrong here.

**Comment by riccardomurri** [comment]
Apparently the documentation is not accurate here; the code requires
that the configuration *file* exists, otherwise the configuration
directory is not scanned at all.  Here's the relevant code (from
`conf.py` lines 133--137)::

        for cfgfile in configfiles[:]:
            cfgdir = cfgfile + '.d'
            if os.path.isfile(cfgfile) and os.path.isdir(cfgdir):
                for fname in os.listdir(cfgdir):
                    # ...

That said, the configuration file can be empty, and all files in the
configuration directory will be read::

        # show a non-empty default config
        $ elasticluster list-templates
        1 cluster templates found in configuration file.

        name:     slurm
        compute nodes: 1
        frontend nodes: 1

        # create empty config file
        $ true > /tmp/cfg

        # put non-empty config into config directory
        $ cp ~/.elasticluster/config /tmp/cfg.d/some.conf
        $ elasticluster -c /tmp/cfg list-templates
        1 cluster templates found in configuration file.

        name:     slurm
        compute nodes: 1
        frontend nodes: 1

In other words, I cannot reproduce the bug you report, that "the files in config.d are completely ignored".

**Comment by riccardomurri** [comment]
This should now be fixed in the "development" branch (commit b3035cf608efb3a36e237096e836d7868b5566c4 or later); can you please test and tell me whether it works as intended?

**Comment by vardaofthevalier** [comment]
Ok, it worked for me just adding a blank config file to the .elasticluster directory.  Now when I run "elasticluster list" I can see all of the clusters that I've created using the config files in .elasticluster/config.d.  

Thanks for the help!
